tutoraspects/templates/aspects/apps/aspects/migrations/alembic/versions/0009_prefer_course_key.py

```diff
@@ -6,208 +8,430 @@
-            if(
-                verb_id = 'https://w3id.org/xapi/acrossx/verbs/evaluated',
-                cast(JSON_VALUE(event_str, '$.result.success') as Bool),
-                false
-            ) as success,
-            JSON_VALUE(event_str, '$.object.definition.interactionType') as interaction_type,
-            if(
-                verb_id = 'https://w3id.org/xapi/acrossx/verbs/evaluated',
-                cast(JSON_VALUE(event_str, '$.object.definition.extensions."http://id.tincanapi.com/extension/attempt-id"') as Int16),
-                0
-            ) as attempts
-        FROM
-            {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_XAPI_TABLE }}
-        WHERE
-            verb_id in (
-                'https://w3id.org/xapi/acrossx/verbs/evaluated',
-                'http://adlnet.gov/expapi/verbs/passed',
-                'http://adlnet.gov/expapi/verbs/asked'
-            );
-        """,
-    "{{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_NAVIGATION_TRANSFORM_MV }}": """
-        CREATE MATERIALIZED VIEW IF NOT EXISTS {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_NAVIGATION_TRANSFORM_MV }}
-        TO {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_NAVIGATION_EVENTS_TABLE }} AS
-        SELECT
-            event_id,
-            emission_time,
-            actor_id,
-            object_id,
-            course_id,
-            org,
-            verb_id,
-            JSON_VALUE(event_str, '$.object.definition.type') AS object_type,
-            -- clicking a link and selecting a module outline have no starting-position field
-            if (
-                object_type in (
-                    'http://adlnet.gov/expapi/activities/link',
-                    'http://adlnet.gov/expapi/activities/module'
-                ),
-                0,
-                cast(JSON_VALUE(
-                    event_str,
-                    '$.context.extensions."http://id.tincanapi.com/extension/starting-position"'
-                ) as Int16)
-            ) AS starting_position,
-            JSON_VALUE(
-                event_str,
-                '$.context.extensions."http://id.tincanapi.com/extension/ending-point"'
-            ) AS ending_point
-        FROM
-            {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_XAPI_TABLE }}
-        WHERE verb_id IN (
-            'https://w3id.org/xapi/dod-isd/verbs/navigated'
-        );
-        """,
-}
+@dataclass
+class MvMigration:
+    table_name: str
+    old_ddl: str
+    new_ddl: str
+    mv_name: str
+    old_mv_query: str
+    new_mv_query: str
+
+
+OLD_ENROLLMENT_DDL = """
+CREATE TABLE IF NOT EXISTS {{ ASPECTS_XAPI_DATABASE }}.{{ ASPECTS_ENROLLMENT_EVENTS_TABLE }} (
+    `event_id` UUID NOT NULL,
+    `emission_time` DateTime64(6) NOT NULL,
+    `actor_id` String NOT NULL,
+    `object_id` String NOT NULL,
+    `course_id` String NOT NULL,
+    `org` String NOT NULL,
+    `verb_id` LowCardinality(String) NOT NULL,
+    `enrollment_mode` LowCardinality(String)
+) ENGINE = MergeTree
```

Comment:
Last request, I think, but just to avoid having to rebuild all of these tables again for #191 can we make these tables `ReplacingMergeTree`? It will save a lot of complexity.